### PR TITLE
TS-55 BUG - Hydration Error

### PR DESCRIPTION
**Pull request by Mark-Ball**
- added a hook so we know when we are on the client and can use browser-only APIs like `window`
- removed `isMobile` package and instead just look at `window.innerWidth` to determine whether to show user name
- fixed a type

### CODE REVIEW

File: src/app/shared/components/NavBar/index.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ const NavBar = async (props: NavbarProps) => {
         </div>
 
         {authenticated
-          ? <UserMenu user={user} />
+          ? <UserMenu user={user!} />
```
Comment (dcarg) [inline src/app/shared/components/NavBar/index.tsx:41]: oooh cool

 _In TypeScript, a postfix ! removes null and undefined from the type of an expression.

This is useful when you know, for reasons outside TypeScript's inference ability, that a variable that "could" be null or undefined actually isn't._

**Comment by dcarg** [comment]
Interesting.

Defs agree using window.innerWidth is practical instead of isMobile.

useIsMounted feels kinda hacky like there should be a better way of addressing this but I spose will do until we think more on it!

**Comment by Mark-Ball** [comment]
> Interesting.
> 
> Defs agree using window.innerWidth is practical instead of isMobile.
> 
> useIsMounted feels kinda hacky like there should be a better way of addressing this but I spose will do until we think more on it!

the solution is directly from next's doc lol, just extracted to its own hook, but i agree it feels weird
https://nextjs.org/docs/messages/react-hydration-error

**Review by dcarg — APPROVED** [review]
👍

**Pull request merged**